simulation/utils/generate_map.py

```diff
@@ -55,82 +57,101 @@ def get_seed(self):
         """Get the seed of the map"""
         return self.seed
 
-    def X(self):
+    def get_X_list(self):
         """Get list of x coordinates"""
-        xs = [point.x for point in self.points_before_interpolation] if not len(self.map_points) else [point[0] for point in self.map_points]
+        if not len(self.map_points):
+            xs = [point.x for point in self.points_before_interpolation]
+        else:
+            xs = [point[0] for point in self.map_points]
+
         return xs
 
-    def Y(self):
+    
+    def get_Y_list(self):
         """Get list of y coordinates"""
-        ys = [point.y for point in self.points_before_interpolation] if not len(self.map_points) else [point[1] for point in self.map_points]
+        if not len(self.map_points):
+            ys = [point.y for point in self.points_before_interpolation]
+        else:
+            ys = [point[1] for point in self.map_points]
+
         return ys
 
+    
     def __len__(self):
         """ Length of map (points, not segments) """
-        return len(self.points_before_interpolation) if not len(self.map_points) else len(self.map_points)
-
-    # ************
+        if len(self.map_points):
+            return len(self.points_before_interpolation)
+        else:
+            return len(self.map_points)
+        
     def __delitem__(self, index):
         if not len(self.map_points):
             del self.points_before_interpolation[index]
         else:
             del self.map_points[index]
 
-    # ************
-    def is_crossing(self, i, j):
+    def are_points_in_clockwise_order(self, A, B, C):
+        return (C.y - A.y) * (B.x - A.x) < (B.y - A.y) * (C.x - A.x)
+
+    def are_lines_intersecting(self, i, j):
```

Comment:
Może zamiast i, j jakieś bardziej znaczące nazwy? Z tego co rozumiem to tu są indeksy odcinków?

Reply:
yup. dla mnie te i i j sa zawsze defaultowo jako indeksy, ale jak to nie jest jasne to zmienie na jakies sensowniejsze nazwy :)

Reply:
Zawsze dobrą praktyką jest nadawanie zmiennym znaczących nazw, zwłaszcza w pythonie w którym zmienna może być wszystkim i nawet sobie wygodnie typu nie sprawdzisz w kodzie. No i póki w c można się posłużyć jakimś i, kiedy go nie używasz w pętli to w pythonie mamy _, który eliminuje takie użycie. W sumie wystarczajce jest zmienienie tego na firstIndex czy secondEdgeIndex czy cokolwiek takiego -> patrząc na to od razu wiesz, że index, do czego służy i potencjalnie że to integer